manifest/build.go

```diff
@@ -21,7 +21,16 @@ func (m *Manifest) Build(dir, appName string, s Stream, cache bool) error {
 		}
 	}
 
+	buildCache := map[string]string{}
+
 	for _, service := range builds {
+		if bc, ok := buildCache[service.Build.Hash()]; ok {
```

Comment:
I think you can't use things with a map as a key. See the "Key types" section of https://blog.golang.org/go-maps-in-action

That being said I think this isnt properly hashing the args map.